subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -72,8 +74,14 @@ public DefaultTransformer(Class<? extends ArtifactTransform> implementationClass
         return outputs;
     }
 
-    private ArtifactTransform newTransformer() {
-        return instantiator.newInstance(implementationClass, parameters.isolate());
+    private ArtifactTransform newTransformer(ArtifactTransformDependencies artifactTransformDependencies) {
+        DefaultServiceRegistry registry = new DefaultServiceRegistry();
+        registry.addProvider(new Object() {
```

Comment:
If we use the `add(Class, T)` method, do we then not require scanning the methods of the anonymous inner class?

Reply:
Good point. We still scan the methods of the `DefaultServiceRegistry` instance, but we can take care of that separately.